### ISSUE TITLE
[Bug] New wallets show same balance and address as first wallet

### ISSUE THREAD

**Issue by chia-rschmidt** [Chia-Network/chia-blockchain#18396]
### What happened?

Any wallets created or imported after the first one show the same wallet address as the first wallet. After syncing the balances shown also match the first wallet in farming mode. In wallet mode the second wallet balances seem to show properly but any wallets after the second have the same issue.

### Version

2.4.3-rc1

### What platform are you using?

Linux

### What ui mode are you using?

GUI

### Relevant log output

```shell
2024-08-03T21:24:06.921 wallet wallet_server              : WARNING  API not ready, ignore request: {'data': '0xa55e6a902987ac0b3a791b12d5098753bc1ba94def824426cd5e934612c332f40011774b00000000000000000000009e6fc108720011774b',
 'id': None,
 'type': 50}
2024-08-03T21:25:28.056 wallet chia.wallet.wallet_node    : ERROR    Disconnected from peer bd27d1a278e6dfdc6794ca5319d5ec619d3310b77f972c2862de25d2a4709af0 host 127.0.0.1
2024-08-03T21:28:23.058 wallet wallet_server              : WARNING  Connection failed to wss://54.185.144.223:58444/ws. Error: 403, message='Invalid response status', url=URL('wss://54.185.144.223:58444/ws')
2024-08-03T21:28:26.911 wallet wallet_server              : WARNING  Connection failed to wss://104.248.202.108:58444/ws. Error: 403, message='Invalid response status', url=URL('wss://104.248.202.108:58444/ws')
2024-08-03T21:30:03.796 full_node full_node_server        : WARNING  Connection failed to wss://136.37.132.123:58444/ws. Error: 403, message='Invalid response status', url=URL('wss://136.37.132.123:58444/ws')
2024-08-03T21:30:58.108 wallet chia.wallet.wallet_node    : ERROR    Error syncing to PeerInfo(_ip=IPv4Address('47.197.8.197'), _port=58444)
Traceback (most recent call last):
  File "chia/wallet/wallet_node.py", line 1217, in new_peak_from_untrusted
  File "chia/wallet/wallet_node.py", line 1226, in long_sync_from_untrusted
  File "chia/wallet/wallet_node.py", line 1355, in fetch_and_update_weight_proof
Exception: weight proof response was none
2024-08-03T21:33:34.212 wallet wallet_server              : WARNING  API not ready, ignore request: {'data': '0x83060727bef3a184ea9ead54f730cccb70b8245b0db8e0fb683eaa7cfe6be6370011776800000000000000000000009e7148887200117768',
 'id': None,
 'type': 50}
2024-08-03T21:36:31.924 full_node full_node_server        : WARNING  cannot connect to 24.22.190.146, duplicate/self connection
2024-08-03T21:36:46.269 wallet wallet_server              : WARNING  Connection failed to wss://54.185.144.223:58444/ws. Error: 403, message='Invalid response status', url=URL('wss://54.185.144.223:58444/ws')
2024-08-03T21:37:22.062 wallet wallet_server              : WARNING  Connection failed to wss://104.248.202.108:58444/ws. Error: 403, message='Invalid response status', url=URL('wss://104.248.202.108:58444/ws')
2024-08-03T21:39:43.421 full_node full_node_server        : WARNING  Banning 75.46.142.121 for 10 seconds
2024-08-03T21:39:43.422 full_node full_node_server        : WARNING  Invalid handshake with peer PeerInfo(_ip=IPv4Address('75.46.142.121'), _port=28744). Maybe the peer is running old software.
2024-08-03T21:42:55.461 wallet chia.wallet.wallet_node    : ERROR    Error syncing to PeerInfo(_ip=IPv4Address('47.197.8.197'), _port=58444)
Traceback (most recent call last):
  File "chia/wallet/wallet_node.py", line 1217, in new_peak_from_untrusted
  File "chia/wallet/wallet_node.py", line 1226, in long_sync_from_untrusted
  File "chia/wallet/wallet_node.py", line 1355, in fetch_and_update_weight_proof
Exception: weight proof response was none
2024-08-03T21:43:15.537 wallet wallet_server              : WARNING  Peer 40.129.58.194 is still banned, not connecting to it
2024-08-03T21:44:55.538 wallet chia.wallet.wallet_node    : ERROR    Error syncing to PeerInfo(_ip=IPv4Address('51.159.109.117'), _port=58444)
Traceback (most recent call last):
  File "chia/wallet/wallet_node.py", line 1217, in new_peak_from_untrusted
  File "chia/wallet/wallet_node.py", line 1226, in long_sync_from_untrusted
  File "chia/wallet/wallet_node.py", line 1355, in fetch_and_update_weight_proof
Exception: weight proof response was none
```

**Comment by chia-rschmidt** [comment]
[TC-charlie.webm](https://github.com/user-attachments/assets/406e906b-df83-4d2f-9ff5-0d03b42b72fa)
[TC-Foxtrot.webm](https://github.com/user-attachments/assets/5169252b-74f1-42b2-97fc-6afe3ec06441)

**Comment by BrandtH22** [comment]
Confirmed this issue in 2.4.3rc1 and effects both CLI and GUI:

CLI response for two wallets:

Wallet 1 (2.4.3rc1):
```
chia wallet show -f 2281896037
Wallet height: 5744166
Sync status: Synced
Balances, fingerprint: 2281896037

Chia Wallet:
   -Total Balance:         0.00017575425 xch (175754250 mojo)
   -Pending Total Balance: 0.00017575425 xch (175754250 mojo)
   -Spendable:             0.00017575425 xch (175754250 mojo)
   -Type:                  STANDARD_WALLET
   -Wallet ID:             1

CAT 2f3a28ba03734b65...:
   -Total Balance:         0.0  (0 mojo)
   -Pending Total Balance: 0.0  (0 mojo)
   -Spendable:             0.0  (0 mojo)
   -Type:                  CAT
   -Asset ID:              2f3a28ba03734b65797df5d5ff0e253606e7fea2e2ee341c39e9a405bce68aa8
   -Wallet ID:             2

CAT 32f9989569a6d457...:
   -Total Balance:         0.0  (0 mojo)
   -Pending Total Balance: 0.0  (0 mojo)
   -Spendable:             0.0  (0 mojo)
   -Type:                  CAT
   -Asset ID:              32f9989569a6d457c5d2e8c72957709c4e1d7adb186f18b0963c93a9eac5c402
   -Wallet ID:             3

CAT 39527b8859d3acfb...:
   -Total Balance:         1.0  (1000 mojo)
   -Pending Total Balance: 1.0  (1000 mojo)
   -Spendable:             1.0  (1000 mojo)
   -Type:                  CAT
   -Asset ID:              39527b8859d3acfb2dcad3956106207d880c2020c3deb0960ab88c337f6dc14a
   -Wallet ID:             4

CAT 0e01b1418f5b8e1c...:
   -Total Balance:         0.001  (1 mojo)
   -Pending Total Balance: 0.001  (1 mojo)
   -Spendable:             0.001  (1 mojo)
   -Type:                  CAT
   -Asset ID:              0e01b1418f5b8e1c8a636ad625793803088ca1ec62a3e8d883abce05eb0101a5
   -Wallet ID:             5
```

Wallet 2 (2.4.3rc1):
```
chia wallet show -f 688295223
Wallet height: 5744124
Sync status: Synced
Balances, fingerprint: 688295223

Chia Wallet:
   -Total Balance:         0.00017575425 xch (175754250 mojo)
   -Pending Total Balance: 0.00017575425 xch (175754250 mojo)
   -Spendable:             0.00017575425 xch (175754250 mojo)
   -Type:                  STANDARD_WALLET
   -Wallet ID:             1

CAT 2f3a28ba03734b65...:
   -Total Balance:         0.0  (0 mojo)
   -Pending Total Balance: 0.0  (0 mojo)
   -Spendable:             0.0  (0 mojo)
   -Type:                  CAT
   -Asset ID:              2f3a28ba03734b65797df5d5ff0e253606e7fea2e2ee341c39e9a405bce68aa8
   -Wallet ID:             2

CAT 32f9989569a6d457...:
   -Total Balance:         0.0  (0 mojo)
   -Pending Total Balance: 0.0  (0 mojo)
   -Spendable:             0.0  (0 mojo)
   -Type:                  CAT
   -Asset ID:              32f9989569a6d457c5d2e8c72957709c4e1d7adb186f18b0963c93a9eac5c402
   -Wallet ID:             3

CAT 39527b8859d3acfb...:
   -Total Balance:         1.0  (1000 mojo)
   -Pending Total Balance: 1.0  (1000 mojo)
   -Spendable:             1.0  (1000 mojo)
   -Type:                  CAT
   -Asset ID:              39527b8859d3acfb2dcad3956106207d880c2020c3deb0960ab88c337f6dc14a
   -Wallet ID:             4

CAT 0e01b1418f5b8e1c...:
   -Total Balance:         0.001  (1 mojo)
   -Pending Total Balance: 0.001  (1 mojo)
   -Spendable:             0.001  (1 mojo)
   -Type:                  CAT
   -Asset ID:              0e01b1418f5b8e1c8a636ad625793803088ca1ec62a3e8d883abce05eb0101a5
   -Wallet ID:             5
```

Wallet 1 (2.4.2):
```
chia wallet show -f 688295223
Wallet height: 5744299
Sync status: Synced
Balances, fingerprint: 688295223

Chia Wallet:
   -Total Balance:         0.0 xch (0 mojo)
   -Pending Total Balance: 0.0 xch (0 mojo)
   -Spendable:             0.0 xch (0 mojo)
   -Type:                  STANDARD_WALLET
   -Wallet ID:             1
```

Wallet 2 (2.4.2):
```
chia wallet show -f 2281896037
Wallet height: 5744124
Sync status: Synced
Balances, fingerprint: 2281896037

Chia Wallet:
   -Total Balance:         0.00017575425 xch (175754250 mojo)
   -Pending Total Balance: 0.00017575425 xch (175754250 mojo)
   -Spendable:             0.00017575425 xch (175754250 mojo)
   -Type:                  STANDARD_WALLET
   -Wallet ID:             1

CAT 2f3a28ba03734b65...:
   -Total Balance:         0.0  (0 mojo)
   -Pending Total Balance: 0.0  (0 mojo)
   -Spendable:             0.0  (0 mojo)
   -Type:                  CAT
   -Asset ID:              2f3a28ba03734b65797df5d5ff0e253606e7fea2e2ee341c39e9a405bce68aa8
   -Wallet ID:             2

CAT 32f9989569a6d457...:
   -Total Balance:         0.0  (0 mojo)
   -Pending Total Balance: 0.0  (0 mojo)
   -Spendable:             0.0  (0 mojo)
   -Type:                  CAT
   -Asset ID:              32f9989569a6d457c5d2e8c72957709c4e1d7adb186f18b0963c93a9eac5c402
   -Wallet ID:             3

CAT 39527b8859d3acfb...:
   -Total Balance:         1.0  (1000 mojo)
   -Pending Total Balance: 1.0  (1000 mojo)
   -Spendable:             1.0  (1000 mojo)
   -Type:                  CAT
   -Asset ID:              39527b8859d3acfb2dcad3956106207d880c2020c3deb0960ab88c337f6dc14a
   -Wallet ID:             4

CAT 0e01b1418f5b8e1c...:
   -Total Balance:         0.001  (1 mojo)
   -Pending Total Balance: 0.001  (1 mojo)
   -Spendable:             0.001  (1 mojo)
   -Type:                  CAT
   -Asset ID:              0e01b1418f5b8e1c8a636ad625793803088ca1ec62a3e8d883abce05eb0101a5
   -Wallet ID:             5
```

Wallet 1 GUI (2.4.3rc1):
![image](https://github.com/user-attachments/assets/d76b0d34-cc4d-4fb5-9c36-b5e895dc8cc8)


Wallet 2 GUI (2.4.3rc1):
![image](https://github.com/user-attachments/assets/8706a15d-c0ff-4df0-9a6b-5b80fa15a1cf)


Wallet 1 GUI (2.4.2):
![image](https://github.com/user-attachments/assets/5b48dd16-24a7-47e3-a443-bf149dffbe6f)


Wallet 2 GUI (2.4.2):
![image](https://github.com/user-attachments/assets/e8181015-3566-436e-94e1-7b4b202a4daa)

**Comment by BrandtH22** [comment]
It seems all wallets are syncing from the first wallet in the keyring. 
I just created a new wallet (no transactions at all) and it synced the same balance and transactions as that first keyring wallet:
![image](https://github.com/user-attachments/assets/c8739e2d-825e-494e-9a21-56bb1ef198bb)

**Comment by BrandtH22** [comment]
This is fixed in #18405 and will be included in 2.4.3RC2
